RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift

```diff
@@ -1180,52 +1229,54 @@ enum FeeValidationOutcome {
 }
 
 extension ReviewedTransaction {
-	var feePayingValidation: FeeValidationOutcome {
-		switch transaction {
-		case .nonConforming, .conforming(.accountDepositSettings):
-			return feePayerSelection.validate
-
-		case let .conforming(.general(generalTransaction)):
-			guard let feePayer = feePayerSelection.selected,
-			      let feePayerWithdraws = generalTransaction.accountWithdraws[feePayer.account.address.address]
-			else {
-				return feePayerSelection.validate
-			}
+	var feePayingValidation: Loadable<FeeValidationOutcome> {
+		feePayer.map { selected in
+			switch transaction {
+			case .nonConforming, .conforming(.accountDepositSettings):
+				return selected.validateBalance(forFee: transactionFee)
+
+			case let .conforming(.general(generalTransaction)):
+				guard let feePayer = selected,
+				      let feePayerWithdraws = generalTransaction.accountWithdraws[feePayer.account.address.address]
+				else {
+					return selected.validateBalance(forFee: transactionFee)
+				}
 
-			let xrdAddress = knownAddresses(networkId: networkId.rawValue).resourceAddresses.xrd
+				let xrdAddress = knownAddresses(networkId: networkId.rawValue).resourceAddresses.xrd
 
-			let xrdTotalTransfer: RETDecimal = feePayerWithdraws.reduce(.zero) { partialResult, resource in
-				if case let .fungible(resourceAddress, source) = resource, resourceAddress == xrdAddress {
-					return (try? partialResult.add(other: source.amount)) ?? partialResult
+				let xrdTotalTransfer: RETDecimal = feePayerWithdraws.reduce(.zero) { partialResult, resource in
+					if case let .fungible(resourceAddress, source) = resource, resourceAddress == xrdAddress {
+						return (try? partialResult.add(other: source.amount)) ?? partialResult
+					}
+					return partialResult
 				}
-				return partialResult
-			}
 
-			let total = xrdTotalTransfer + feePayerSelection.transactionFee.totalFee.lockFee
+				let total = xrdTotalTransfer + transactionFee.totalFee.lockFee
 
-			guard feePayer.xrdBalance >= total else {
-				// Insufficient balance to pay for withdraws and transaction fee
-				return .insufficientBalance
-			}
+				guard feePayer.xrdBalance >= total else {
+					// Insufficient balance to pay for withdraws and transaction fee
+					return .insufficientBalance
+				}
 
-			return .valid
+				return .valid
+			}
 		}
 	}
 }
 
-extension FeePayerSelectionAmongstCandidates {
-	var validate: FeeValidationOutcome {
+extension FeePayerCandidate? {
```

Comment:
perhaps straining for type-checker to spell `FeePayerCandidate?`, maybe `extension Optional<FeePayerCandidate>` is faster? more clear as well, I think I prefer it.